arelle/RuntimeOptions.py

```diff
@@ -0,0 +1,149 @@
+    packages: Optional[str] = None,
+    parameterSeparator: Optional[str] = None,
+    parameters: Optional[str] = None,
+    password: Optional[str] = None,
+    plugins: Optional[str] = None,
+    preFile: Optional[str] = None,
+    proxy: Optional[str] = None,
+    relationshipCols: Optional[int] = None,
+    roleTypesFile: Optional[str] = None,
+    rssReport: Optional[str] = None,
+    rssReportCols: Optional[int] = None,
+    showEnvironment: Optional[bool] = None,
+    showOptions: Optional[bool] = None,
+    skipDTS: Optional[bool] = None,
+    skipLoading: Optional[bool] = None,
+    statusPipe: Optional[str] = None,
+    tableFile: Optional[str] = None,
+    testReport: Optional[str] = None,
+    testReportCols: Optional[int] = None,
+    testcaseResultOptions: Optional[str] = None,
+    testcaseResultsCaptureWarnings: Optional[bool] = None,
+    timeVariableSetEvaluation: Optional[bool] = None,
+    uiLang: Optional[str] = None,
+    username: Optional[str] = None,
+    utrUrl: Optional[str] = None,
+    utrValidate: Optional[bool] = None,
+    validate: Optional[bool] = None,
+    validateEFM: Optional[bool] = None,
+    validateEFMCalcTree: Optional[bool] = None,
+    validateHMRC: Optional[bool] = None,
+    validateTestcaseSchema: Optional[bool] = None,
+    versReportFile: Optional[str] = None,
+    viewArcrole: Optional[bool] = None,
+    viewer_feature_review: Optional[bool] = False,
+    viewFile: Optional[str] = None,
+    webserver: Optional[str] = None,
+    xdgConfigHome: Optional[str] = None
+
+    def __post_init__(self, pluginOptions: Optional[dict[str, RuntimeOptionValue]]) -> None:
+        if pluginOptions:
+            existingBaseOptions = sorted(
+                optionName
+                for optionName in pluginOptions.keys()
+                if hasattr(self, optionName)
+            )
+            if existingBaseOptions:
+                raise RuntimeOptionsException(_('Provided plugin options already exist as base options {}'.format(existingBaseOptions)))
+            for optionName, optionValue in pluginOptions.items():
+                setattr(self, optionName, optionValue)
+        if (self.entrypointFile is None and ((not self.proxy) and (not self.plugins) and
+                                             (not any(pluginOption for pluginOption in pluginOptions.keys())) and
+                                             (not hasWebServer or self.webserver is None))):
+            raise RuntimeOptionsException(_('Incorrect arguments'))
+        if hasWebServer and self.webserver and any((
+                self.entrypointFile, self.importFiles, self.diffFile, self.versReportFile,
+                self.factsFile, self.factListCols, self.factTableFile, self.factTableCols, self.relationshipCols,
+                self.conceptsFile, self.preFile, self.tableFile, self.calFile, self.dimFile, self.anchFile, self.formulaeFile, self.viewArcrole, self.viewFile,
+                self.roleTypesFile, self.arcroleTypesFile
```

Comment:
The line length is kind of long on a couple of these. I'd break those across multiple lines to make them shorter.